doc/examples/features_detection/plot_scattering_classification.py

```diff
@@ -0,0 +1,259 @@
+"""
+===========================================
+Scattering features for supervised learning
+===========================================
+
+This example shows how to use the scattering features
+for image classification. More specifically, we compute
+the scattering vectors from the MNIST and CIFAR10 databases
+and show how to successfully
+classify their images using Support Vector Machine (SVM)
+with Gaussian kernels. These two
+databases are widely used in research to compare the quality
+of image representations and
+classification methods. We provide the parameters obtained by
+crossvalidation to reproduce the results obtained
+in [1]_ and [2]_.
+Scattering features have shown to outperform any other 'non-learned'
+image representations for image classification
+tasks [2]_.
+Structurally, they are very similar to deep learning representations,
+with fixed filters, Morlet filters in this implementation.
+Since the scattering transform is based on the
+Discrete Wavelet transform (DWT), it is stable to deformations.
+It is also invariant to small translations. For more details on
+its mathematical properties and exact definition,
+see the example *Scattering Features* or check [1]_ and [2]_.
+
+ **Supervised Learning: Image Classification**
+The first task in supervised image classification is to collect a
+set of images tagged with a class. In case of the MNIST
+database we have pictures of digits, and the class is the
+corresponding number. Then, for every image
+we compute a feature vector which, in our case, is the
+scattering vector. Once we have the complete
+database, we 'train' a classifier, meaning, we find the
+best parameters that allow to correctly classify the images.
+A widely used classifier is Support Vector Machine (SVM),
+which searches the high dimensional plane
+that allows to better separate each pair of classes,
+see [3]_ for more information.
+Here, we show how to reproduce the results obtained
+with the scattering transform, using SVM (with Gaussian kernels)
+first presented in the following papers:
+
+- **MNIST database** :
+ The results were first presented in [1]_ Table 4. Using this
+ implementation and a 10000 image test set, we were able to obtain, the following
```

Comment:
The thing is that for some lines within a comment, or that become afterward text, changing the length would change the format afterwards in the example. So this is a middle point